articles/actions.md

```diff
@@ -0,0 +1,437 @@
+Following the goal request, the client makes an asynchronous request for the result.
+The user defined method publishes feedback to the action client as it executes the goal.
+Ultimately, the user defined method populates a result message that is used as part of the result response.
+
+![Action Interaction Example 0](../img/actions/interaction_example_0.png)
+
+#### Example 2
+
+This example is almost identical to the first, but this time the action  client requests for the goal to be canceled mid-execution.
+Note that the user defined method is allowed to perform any shutdown operations after the cancel request before returning with the cancellation result.
+
+![Action Interaction Example 1](../img/actions/interaction_example_1.png)
+
+#### Example 3
+
+Here is a more complex example involving multiple goals.
+
+![Action Interaction Example 2](../img/actions/interaction_example_2.png)
+
+### Topic and Service Name Examples
+
+Topic and service names for actions will include a token `_action`.
+The leading underscore makes the name hidden, and the combined text allows tools to identify topics and services used by actions.
+Topic and Service names will be generated by prefixing the action name to one of `/_action/status`, `/_action/feedback`, `/_action/get_result`, `/_action/cancel_goal`, or `/_action/send_goal`.
+The resulting topic or service name is expanded to a fully qualified name as usual.
+
+#### Example: fully qualified action name
+Given:
+* Action name: `/action/name`
+* Namespace: `/name/space`
+* Node name: `nodename`
+
+Topic Names:
+
+* `/action/name/_action/status`
+* `/action/name/_action/feedback`
+
+Service Names:
+
+* `/action/name/_action/send_goal`
+* `/action/name/_action/cancel_goal`
+* `/action/name/_action/get_result`
+
+#### Example: relative action name
+Given:
+* Action name: `action/name`
+* Namespace: `/name/space`
+* Node name: `nodename`
+
+Topic Names:
+
+* `/name/space/action/name/_action/status`
+* `/name/space/action/name/_action/feedback`
+
+Service Names:
+
+* `/name/space/action/name/_action/send_goal`
+* `/name/space/action/name/_action/cancel_goal`
+* `/name/space/action/name/_action/get_result`
+
+#### Example: private action name
+Given:
+* Action name: `~/action/name`
+* Namespace: `/name/space`
+* Node name: `nodename`
+
+Topic Names:
+
+* `/name/space/nodename/action/name/_action/status`
+* `/name/space/nodename/action/name/_action/feedback`
+
+Service Names:
+
+* `/name/space/nodename/action/name/_action/send_goal`
+* `/name/space/nodename/action/name/_action/cancel_goal`
+* `/name/space/nodename/action/name/_action/get_result`
+
+## Bridging between ROS 1 and ROS 2
+
+TODO
+
+## Alternatives
+
+These alternative approaches to actions in ROS 2 were considered.
+
+### Actions in rmw
+
+An alternative to using services and topics is to implement actions in the rmw layer.
+This would enable using middleware specific features better suited actions.
```

Comment:
`better suited actions` -> `better suited to actions`.

Reply:
Fixed: f4b59fa